### ISSUE TITLE
[TB] Enhance Responsiveness of ThemeBuilder Application on Mobile Devices

### ISSUE THREAD

**Issue by Sweetdevil144** [finos/a11y-theme-builder#855]
## Problem/Concern
<img width="501" alt="Screenshot 2024-05-06 at 2 19 02 AM" src="https://github.com/finos/a11y-theme-builder/assets/117591942/dce97505-6fa6-4e81-b670-98e8a161f6d2">

Non responsiveness exhibited in small window view.
* This is one of the deployments where I found the issue : [link to problem content](https://deploy-preview-826--glistening-gecko-6b417a.netlify.app/)

## Proposed Solution
- Utilization of Hamburger menu for sidebar.
- Utilization of Flex and Grid Layouts wherever required.
-Introduce media queries to adjust text sizes and layout dynamically based on the screen size.

**Comment by aaronreed708** [comment]
Thanks for opening the issue @Sweetdevil144!  We haven't spent much time working on responsiveness, yet, since it is a design tool that isn't likely to be used on a very small screen.  Probably nothing smaller than a tablet.  With that in mind, do you have a suggested list of things that we could fix to assure a good user experience on a tablet?  So that we can prioritize the work?

**Comment by Sweetdevil144** [comment]
Here are a few things we could prioritize:

- Optimizing the sidebar: Implementing a collapsible hamburger menu can help free up space and improve navigation.
- Flexible layouts: Using Flexbox and Grid can make the UI more adaptable, ensuring elements scale properly on various tablet sizes.
- Media queries: Adjusting basic UI components like buttons, text sizes, and spacing can greatly improve readability and interaction on smaller devices.

**Comment by kburk1997** [comment]
As discussed in the 5/16 meeting, this is not exclusively limited to mobile screens - this is an accessibility issue as well.

Web Content Accessibility Guidelines (WCAG) has a success criterion that mentions this - [1.4.10: Reflow](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html):

> Content can be presented without loss of information or functionality, and without requiring scrolling in two dimensions for:
> 
> -Vertical scrolling content at a width equivalent to 320 [CSS pixels](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html#dfn-css-pixel);
> -Horizontal scrolling content at a height equivalent to 256 [CSS pixels](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html#dfn-css-pixel).
> 
> Except for parts of the content which require two-dimensional layout for usage or meaning.

**Comment by Sweetdevil144** [comment]
Hey @kburk1997 , you're right. This is an accessibility issue in Whole as per the WCAG guidelines as well. Enhancing the `Views` may prove beneficial in fixing this issue.

**Comment by aaronreed708** [comment]
You are all making great points and good suggestions. @Sweetdevil144 would you like me to assign this issue to you?

**Comment by Sweetdevil144** [comment]
Yeah sure. I'd love to work on this. I'm wrapping up the `test Enhancement` Issue. So, I'll start with this parallely too.

**Comment by AradhyaDixit18** [comment]
@aaronreed708 @Sweetdevil144  noticed the same issue related to responsiveness . Can i help for some improvements? If yes then please assign me this task.

**Comment by Sweetdevil144** [comment]
Hey @AradhyaDixit18 . Sure go ahead!! I'm hell busy right now it's open for peeps. You'd be a great help 😁

**Comment by AradhyaDixit18** [comment]
@Sweetdevil144 okk then . I'll try to give my best.

**Comment by aaronreed708** [comment]
Thanks for volunteering! I’m on vacation this week, but happy to help if you have any questions next week.


Sent from my iPhone

On Jun 29, 2024, at 4:20 PM, Aradhya DIxit ***@***.***> wrote:

﻿

@Sweetdevil144<https://github.com/Sweetdevil144> okk then . I'll try to give my best.

**Comment by AradhyaDixit18** [comment]
@aaronreed708  Sir i just fixed the main landing page and made it responsive and created a PR #949 please review this and merge it also review the PR #946 where i have fixed the hex value and added a dynamic hex value i also created a separate issue for it please have a look

**Comment by aaronreed708** [comment]
Hi @AradhyaDixit18!  Thank you for your contribution!  I made some comments in #946 already so I'll keep those comments separate from those related to this issue.

I like the approach that you took in the first PR https://github.com/finos/a11y-theme-builder/pull/949 and the visual that you were able to achieve.  I'll put comments in the PR for PR-specific issues that I see.
